crates/bevy_asset/src/server/info.rs

```diff
@@ -264,9 +268,12 @@ impl AssetInfos {
         self.infos.get_mut(&id)
     }
 
-    pub(crate) fn get_path_handle(&self, path: AssetPath) -> Option<UntypedHandle> {
-        let id = *self.path_to_id.get(&path)?;
-        self.get_id_handle(id)
+    pub(crate) fn get_path_handles(&self, path: AssetPath) -> Vec<UntypedHandle> {
```

Comment:
On a scale from 0 to 10, how impossible is it to return an `impl Iterator<Item=UntypedHandle> + '_`?

Reply:
For `AssetInfos::get_path_handles`? Easy.
For `AssetServer::get_handles_untyped`? I actually think impossible. It hides the `AssetInfo` behind a `RwLock`. So to return an `impl Iterator`, it needs to get a `RwLockGuard`, build the iterator, and then return both. The `RwLockGuard` can't outlive the iterator if it was included (e.g., a new struct owning the inner iterator and the guard), and there's nowhere sensible to store it.

If the `RwLock` was replaced with something with a little more control (e.g., somehow store a `&AssetInfos` only when the read lock should be held), but that's getting pretty involved pretty fast.